auth/external.go

```diff
@@ -0,0 +1,63 @@
+package auth
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+
+	"github.com/fabiolb/fabio/config"
+)
+
+type external struct {
+	endpoint          string
+	appendAuthHeaders []string
+	setAuthHeaders    []string
+}
+
+func newExternalAuth(cfg config.ExternalAuth) (AuthScheme, error) {
+	return &external{
+		endpoint:          cfg.Endpoint,
+		appendAuthHeaders: cfg.AppendAuthHeaders,
+		setAuthHeaders:    cfg.SetAuthHeaders,
+	}, nil
+}
+
+func (b *external) Authorized(request *http.Request, response http.ResponseWriter) AuthDecision {
+	client := &http.Client{CheckRedirect: func(req *http.Request, via []*http.Request) error { return http.ErrUseLastResponse }}
+
+	authRequest, err := http.NewRequest(request.Method, b.endpoint+request.URL.RequestURI(), bytes.NewReader(nil))
+	if err != nil {
+		log.Println("[ERROR] Can't make auth external request value:", err.Error())
+		return unauthorized()
+	}
+	authRequest.Header = request.Header
+	authRequest.Host = request.Host
+
+	resp, err := client.Do(authRequest)
+	if err != nil {
+		log.Println("[ERROR] External request error:", err.Error())
+		return unauthorized()
+	}
+
```

Comment:
resp.Body is never drained or closed.  if err is nil, body needs to be dealt with.